python/cuvs/cuvs/neighbors/all_neighbors/all_neighbors.pyx

```diff
@@ -0,0 +1,384 @@
+                    f"Metric conflict: AllNeighborsParams metric '{metric}' "
+                    f"does not match IVF-PQ metric '{ivf_pq_metric}'. Please "
+                    f"ensure both use the same metric."
+                )
+
+        if nn_descent_params is not None:
+            if not isinstance(nn_descent_params, NNDescentIndexParams):
+                raise TypeError(
+                    "nn_descent_params must be an instance of "
+                    "cuvs.neighbors.nn_descent.IndexParams"
+                )
+
+            # Check metric consistency
+            nn_descent_metric = nn_descent_params.metric
+            if nn_descent_metric != metric:
+                raise ValueError(
+                    f"Metric conflict: AllNeighborsParams metric '{metric}' "
+                    f"does not match NN-Descent metric '{nn_descent_metric}'. "
+                    f"Please ensure both use the same metric."
+                )
+
+        # Store references to prevent garbage collection
+        self._ivf_pq_params = ivf_pq_params
+        self._nn_descent_params = nn_descent_params
+
+        # Set algorithm-specific parameter pointers
+        if ivf_pq_params is not None:
+            self.params.ivf_pq_params = (
+                <cuvsIvfPqIndexParams_t><size_t>ivf_pq_params.get_handle()
+            )
+        else:
+            self.params.ivf_pq_params = <cuvsIvfPqIndexParams_t>NULL
+
+        if nn_descent_params is not None:
+            self.params.nn_descent_params = (
+                <cuvsNNDescentIndexParams_t><size_t>(
+                    nn_descent_params.get_handle()
+                )
+            )
+        else:
+            self.params.nn_descent_params = <cuvsNNDescentIndexParams_t>NULL
+
+    def get_handle(self):
+        """Get a pointer to the underlying C object."""
+        return <size_t>&self.params
+
+    @property
+    def algo(self):
+        """Algorithm used for local k-NN graph building."""
+        if self.params.algo == CUVS_ALL_NEIGHBORS_ALGO_BRUTE_FORCE:
+            return "brute_force"
+        elif self.params.algo == CUVS_ALL_NEIGHBORS_ALGO_IVF_PQ:
+            return "ivf_pq"
+        elif self.params.algo == CUVS_ALL_NEIGHBORS_ALGO_NN_DESCENT:
+            return "nn_descent"
+        else:
+            return self.params.algo
+
+    @property
+    def overlap_factor(self):
+        """Number of clusters each point is assigned to."""
+        return self.params.overlap_factor
+
+    @property
+    def n_clusters(self):
+        """Number of clusters/batches to partition the dataset into."""
+        return self.params.n_clusters
+
+    @property
+    def metric(self):
+        """Distance metric used for graph construction."""
+        # Reverse lookup in DISTANCE_TYPES
+        for name, value in DISTANCE_TYPES.items():
+            if value == self.params.metric:
+                return name
+        return self.params.metric
+
+
+# @auto_sync_snmg_resources
```

Comment:
Make sure we uncomment this. 

I don't think we want to assume snmg here all the time, right? When we're not using snmg, we want to just have all-neighbors function as a single gpu.